### PR TITLE
Adds test parachain adder collator 

### PR DESCRIPTION
**Pull request by bkchr**
Based on: https://github.com/paritytech/polkadot/pull/1795

### CODE REVIEW

File: parachain/test-parachains/adder/collator/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		}
+	}
+
+	/// Get the SCALE encoded genesis head of the adder parachain.
+	pub fn get_genesis_head(&self) -> Vec<u8> {
+		self.state.lock().unwrap().genesis_state.encode()
+	}
+
+	/// Get the validation code of the adder parachain.
+	pub fn get_validation_code(&self) -> &[u8] {
```
Comment (pepyakin) [inline parachain/test-parachains/adder/collator/src/lib.rs:92]: nit: getters don't [get](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter) `get` in front

File: parachain/test-parachains/adder/collator/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+/// The collator of the adder parachain.
+pub struct Collator {
+	state: Arc<Mutex<State>>,
+}
+
+impl Collator {
+	/// Create a new
+	pub fn init() -> Self {
```
Comment (pepyakin) [inline parachain/test-parachains/adder/collator/src/lib.rs:80]: nit: why not `new`?

File: parachain/test-parachains/adder/collator/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	}
+}
+
+/// The collator of the adder parachain.
+pub struct Collator {
+	state: Arc<Mutex<State>>,
+}
+
+impl Collator {
+	/// Create a new
```
Comment (pepyakin) [inline parachain/test-parachains/adder/collator/src/lib.rs:79]: seems to be unfinished.

We could write like:

```suggestion
	/// Create a new collator instance with the state initialized as genesis.
```

**Review by pepyakin — COMMENTED** [review]
nits. Also seems like the branch was merged in the end!

**Comment by coriolinus** [comment]
Fix for tests [here](https://github.com/paritytech/polkadot/compare/master...prgn-adder-collator).

File: parachain/test-parachains/adder/collator/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply parachain/test-parachains/adder/collator/src/lib.rs:92]: Yeah I know, however sometimes I feel like I need them! xD

File: parachain/test-parachains/adder/collator/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply parachain/test-parachains/adder/collator/src/lib.rs:80]: I feeled like `init` ;)

File: parachain/test-parachains/adder/collator/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply parachain/test-parachains/adder/collator/src/lib.rs]: Done

**Comment by montekki** [comment]
a `CollateOn` message has to be sent to the collator protocol in order to properly init it, so there is a little [patch](https://gist.github.com/montekki/89f0d6bbf7120886cdb82a34a445cc08) i've been testing this branch with today.

**Review by rphmeier — APPROVED** [review]

**Review by montekki — APPROVED** [review]
A `README.md` in the adder dir describing how to run the whole setup would be super useful.

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**